lib/iris/tests/unit/common/lenient/test_Lenient.py

```diff
@@ -4,179 +4,166 @@
 # See LICENSE in the root of the repository for full licensing details.
 """Unit tests for the :class:`iris.common.lenient.Lenient`."""
 
-# Import iris.tests first so that some things can be initialised before
-# importing anything else.
-import iris.tests as tests  # isort:skip
+import pytest
 
-from unittest.mock import sentinel
+from iris.common.lenient import _LENIENT, _LENIENT_PROTECTED, Lenient
 
-from iris.common.lenient import _LENIENT, Lenient
 
+@pytest.fixture()
+def lenient():
+    # setup
+    state = {key: _LENIENT.__dict__[key] for key in _LENIENT_PROTECTED}
+    # call
+    yield Lenient()
+    # teardown
+    for key, value in state.items():
+        _LENIENT.__dict__[key] = value
```

Comment:
@pp-mo As I suspected ... `unittest` gives us this `setup` and `teardown` behaviour for free.

Very interesting to know and be aware of ... for reference see https://docs.pytest.org/en/latest/how-to/fixtures.html#yield-fixtures-recommended